osu.Game/Overlays/Settings/Sections/Input/KeyBindingRow.cs

```diff
@@ -583,8 +607,16 @@ public void UpdateKeyCombination(KeyCombination newCombination)
                 if (KeyBinding.RulesetName != null && !RealmKeyBindingStore.CheckValidForGameplay(newCombination))
                     return;
 
+                if (KeyBinding.RulesetName != null && keyCombinationExists(newCombination) && !newCombination.Equals(KeyBinding.KeyCombination))
+                {
+                    box.FlashColour(colours.RedDark, 1000, Easing.In);
+                    isErroring = true;
```

Comment:
Why is this setting the private rather than the property? And why is the property having a public `set`ter in the first place? Feels prone to error.